### PR TITLE
Add js folder test

### PR DESCRIPTION
**Pull request by maoznir**
Put all node tests under spec/env

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,7 +13,8 @@
     "build": "parallel-webpack && npm run bundlewatch",
     "bundlewatch": "bundlewatch --config ./bundlewatch.config.js",
     "doc": "jsdoc -c jsdoc-conf.json",
-    "test": "karma start --single-run --browsers=ChromeHeadless",
+    "pretest": "npm run build && npm run copybuilds",
```
Comment (asisayag2) [inline package.json:16]: @maoznir , is the pretest part of this change?

File: package.json [hunk 2]
```diff
@@ -13,7 +13,8 @@
...
+    "test": "jasmine test/spec/env/*-spec.js && karma start --single-run --browsers=ChromeHeadless",
```
Comment (asisayag2) [inline package.json:17]: @maoznir is jasmine the only way to do that?
Did we considered git's hooks? https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks

Comment (maoznir) [reply package.json:17]: That's also an option

File: package.json (hunk 1 above)
Comment (maoznir) [reply package.json:16]: yes

File: package.json (hunk 2 above)
Comment (asisayag2) [reply package.json:17]: What so you find better?

File: package.json (hunk 1 above)
Comment (asisayag2) [reply package.json:16]: Why is that change requires a pretest?

File: package.json (hunk 2 above)
Comment (patrick-tolosa) [reply package.json:17]: I think hooks can run this test, but we should still have it as a separate test as here.

Personally I prefer the solution we have in this PR.

File: package.json (hunk 1 above)
Comment (maoznir) [reply package.json]: I've refactored:
1. remove pretest
2. update test script to test folder structure
3. update prepublish script to verify packages, where it's actually needed

Comment (maoznir) [reply package.json]: refactored again due to travis breaking, please see updated changes

**Pull request merged**